src/main/java/se/munhunger/idp/model/ErrorMessage.java

```diff
@@ -7,7 +7,7 @@
  * @author Marcus Münger
  */
 @ApiModel(value = "An error message describing something that went wrong")
-public class ErrorMessage {
+public class ErrorMessage extends Exception {
```

Comment:
Rätt tänkt, men tror det är bättre att försöka hålla alla exceptions specifika så att man vet vad som har hänt.
Det här ju indirekt samma sak som att slänga Exception vilket ger väldigt lite info om vad som gick fel.

Reply:
Plockar bort arvet

Reply:
Istället för att det skall fungera har jag satt ditt "extends Thowable" för att man ska kunna kasta ErrorMessage...

Reply:
NEJ!
det är MYCKET värre att slänga Throwable... poängen jag försökte få fram är att du inte får någon vettig info om vad som går fel och du får ännu mindre info om du sätter throwable.

Tänk dig följande:
try{
//blaha
}
catch(ErrorMessage e) {
switch(e.getMessage): {
case "fel1":
//blaha
case "fel2":
//blaha
}
i ditt fall måste man ju kolla i innehållet och sen sträng matcha vilket kan vara farligt.
du vill istället ha något i stil med:
try{
//blaha
}
catch(SpecificError1 e) {
//blaha
}
catch(CompletelyDifferentError e) {
//blaha
}

Reply:
Vilket jag har gjort i "rest.User", "services.UserService"

Och extendar jag inte "Exception" får jag kompilerings fel. 

Söker jag exempel på hur man skapar egna exceptions extendar man i alla exempel exception:
https://www.google.se/search?client=firefox-b-ab&dcr=0&ei=_l18Wo74Ocj8sQHTr5NY&q=how+to+create+your+own+exception+classes+without+extending+exception+java&oq=how+to+create+your+own+exception+classes+without+extending+exception+java&gs_l=psy-ab.3...8756.9359.0.9482.5.5.0.0.0.0.137.493.3j2.5.0....0...1c.1.64.psy-ab..0.0.0....0.bfLLdOFYi0I


eller throwable dessutom använder vi ju:

Optional.ofNullable(getUser(user.getUsername()).isPresent()).orElseThrow(
                    () -> new NotInDatabaseException("User do not exist",
                            "User with username: " + user.getUsername() + " do not exist in DB"));
Extendar vi inte Exception eller throwable kan vi inte använda oss utav ".orElseThrow()" då det inte är kompatibelt men något annat typ av objekt.....

Reply:
någon annan typ*